### ISSUE TITLE
 is not assignable to parameter of type Omit

### ISSUE THREAD

**Issue by craigcosmo** [isaacHagoel/svelte-dnd-action#445]
<img width="761" alt="Screenshot 2023-04-20 at 7 53 58 AM" src="https://user-images.githubusercontent.com/538657/233230457-5096b09d-1922-4c32-85f7-394976489333.png">

out  of the box, following the github installation, I got this error.

**Comment by isaacHagoel** [comment]
Have you checked the section about typescript in the readme?

**Comment by craigcosmo** [comment]
checking now. My sveltekit don't have `global.d.ts` it has `app.d.ts`, are they the same ?

the app.d.ts is like this

```
import type { PrismaClient } from '@prisma/client'

declare global {
	namespace App {
		// interface Error {}
		// interface Locals {}
		// interface PageData {}
		// interface Platform {}
	}
	var prisma: PrismaClient
}

export {}
```

**Comment by isaacHagoel** [comment]
It should work the same for svelte and sveltekit

**Comment by craigcosmo** [comment]
@DeveloperMarius can you post your tsconfig.ts ?

**Comment by DeveloperMarius** [comment]
Hey,

I added nothing related to this library into the config.

~ Marius

**Comment by camargo** [comment]
I ran into this when upgrading from `eslint-plugin-svelte3` to `eslint-plugin-svelte`. Here is what I needed to update:

 ```ts
 declare namespace svelteHTML {
    interface HTMLAttributes<T> {
      'on:consider'?: (event: CustomEvent<DndEvent<ItemType>> & { target: EventTarget & T }) => void;
      'on:finalize'?: (event: CustomEvent<DndEvent<ItemType>> & { target: EventTarget & T }) => void;
    }
  }
```

Following this: https://github.com/sveltejs/language-tools/blob/master/docs/preprocessors/typescript.md#im-getting-deprecation-warnings-for-sveltejsx--i-want-to-migrate-to-the-new-typings

Maybe we should update the README here?

**Comment by isaacHagoel** [comment]
Thanks and sure. Would you like to submit a pull request so you become a
contributor to this repo or would you like me to update it?

**Comment by Holf** [comment]
I have just managed to solve the same problem, by looking at the comments on this issue and combining that with what is written in the [ReadMe](https://github.com/isaacHagoel/svelte-dnd-action#typescript).

However, perhaps the instructions in the ReadMe could be expanded a bit? As explained here https://github.com/isaacHagoel/svelte-dnd-action/issues/445#issuecomment-1517505075:

* The file where this needs to be added could be `app.d.ts` (which arrived out-of-the-box when I used the SvelteKit scaffolding) or `global.d.ts` (which is, I think, more typical of a TypeScript application).
* The extra type declarations need to be nested _inside_ the existing `declare global { ... }` block.

This wasn't clear to me until I found this issue.

Perhaps this is obvious to a seasoned TypeScript expert, but that is not me (yet! I'm starting to have lots of fun with it).

This component has been just awesome in every other respect. 😎
